tornado/simple_httpclient.py

```diff
@@ -491,9 +491,17 @@ def headers_received(self, first_line, headers):
             self.request.header_callback('\r\n')
 
     def _should_follow_redirect(self):
-        return (self.request.follow_redirects and
+        if not (self.request.follow_redirects and
                 self.request.max_redirects > 0 and
-                self.code in (301, 302, 303, 307, 308))
+                self.code in (301, 302, 303, 307, 308)):
+            return False
+        # http://www.w3.org/Protocols/rfc2616/rfc2616-sec10.html#sec10.3.4
+        # According to the spec, the temporary URI may not be specified when the
+        # request method was HEAD.
+        # TODO: Unless the request method was HEAD, give a warning or raise an
+        # exception when the temporary URI is not specified?
+        redirect_location = self.headers.get("Location")
+        return (redirect_location is not None)
```

Comment:
The actual code change is fine, but I think I'd add it in to the same return expression above (`and self.headers.get('Location') is not None`) instead of making this method have two separate return statements.